contributed_definitions/nyaml/NXinstrument.yaml

```diff
@@ -16,26 +16,41 @@ NXinstrument(NXobject):
     \@short_name:
       doc: |
         short name for instrument, perhaps the acronym
-  energy_resolution(NX_FLOAT):
-    unit: NX_ENERGY
+  energy_resolution(NXresolution):
     doc: |
-      Energy resolution of the experiment (FWHM or gaussian broadening)
-  momentum_resolution(NX_FLOAT):
-    unit: NX_WAVENUMBER
+      Energy resolution of the experiment (FWHM or gaussian broadening).
+    physical_quantity:
+      enumeration: [energy]
+    resolution(NXdata):
+      unit: NX_ENERGY
+  momentum_resolution(NXresolution):
     doc: |
-      Momentum resolution of the experiment (FWHM)
-  angular_resolution(NX_FLOAT):
-    unit: NX_ANGLE
+      Momentum resolution of the experiment (FWHM or gaussian broadening).
+    physical_quantity:
+      enumeration: [momentum]
+    resolution(NXdata):
+      unit: NX_WAVENUMBER
+  angular_resolution(NXresolution):
     doc: |
-      Angular resolution of the experiment (FWHM)
-  spatial_resolution(NX_FLOAT):
-    unit: NX_LENGTH
+      Angular resolution of the experiment (FWHM or gaussian broadening).
+    physical_quantity:
+      enumeration: [angle]
+    resolution(NXdata):
+      unit: NX_ANGLE
+  spatial_resolution(NXresolution):
     doc: |
       Spatial resolution of the experiment (Airy disk radius)
-  temporal_resolution(NX_FLOAT):
-    unit: NX_TIME
+    physical_quantity:
+      enumeration: [length]
+    resolution(NXdata):
+      unit: NX_LENGTH
+  temporal_resolution(NXresolution):
     doc: |
       Temporal resolution of the experiment (FWHM)
+    physical_quantity:
+      enumeration: [time]
+    resolution(NXdata):
+      unit: NX_TIME
```

Comment:
I never liked it too much to put these explicit resolutions into NXinstrument, because in the originial NXinstrument, there are apart from this essentially only unnamed base classes cited (to provide the whole branch of possibilities). So this kind of goes against the original design of NXinstrument.

Reply:
I kind of agree here. We only do this to set specific pre-named resolutions and I like the approach of naming these resolutions so everyone would use the same. I think eventually we should build a baseclass inheritance scheme for this, i.e., `NXspatial_resolution`, `NXtemporal_resolution` as subclasses of `NXresolution`. That way we make them reusable and identifiable but don't have to restate them in `NXinstrument`. Would this approach work better for you?

Reply:
Another question is whether to keep the explicitly named resolutions in `NXelectronanalyser`. I suggest to keep them there until we have the suggested baseclass inheritance scheme for this.

Reply:
Explicit resolutions names have been removed from `base_classes/NXinstrument`. I also saw that there was an additional  `contributed_definitions/NXinstrument`, this one was removed.

Reply:
Left explicit resolutions names them in `NXelectronanalyser `to keep the eixsting descriptions. These can be later moved to standalone base classes (`NXspatial_resolution`, `NXtemporal_resolution`, ...) as subclasses of `NXresolution`.